khi_robot_control/src/khi_robot_krnx_driver.cpp

```diff
@@ -334,58 +334,16 @@ bool KhiRobotKrnxDriver::activate( const int cont_no, JointData *joint )
     }
 
     /* Load RTC param */
-    snprintf( param, sizeof(param), "%s/param/rb_rtc_%s.pg", KHI_ROBOT_CONTROL_DIR, p_rb_tbl[cont_no]->robot_name.c_str() );
-    return_code = krnx_Load( cont_no, param );
-    retKrnxRes( cont_no, "krnx_Load", return_code );
-
-    /* Load HOME param */
-    snprintf( param, sizeof(param), "%s/param/rb_rtc_home.pg", KHI_ROBOT_CONTROL_DIR );
-    if ( ( fp = fopen( param, "w" ) ) != NULL) {
-        /* Update HOME position */
-        for ( int ano = 0; ano < robot_info[cont_no].arm_num; ano++ )
-        {
-            if ( !getCurMotionData( cont_no, ano, &motion_data ) ) { return false; }
-
-            for ( int jt = 0; jt < joint->joint_num; jt++ )
-            {
-                memcpy( &p_rb_tbl[cont_no]->arm_tbl[ano].jt_tbl[jt].home, &motion_data.ang_ref[jt], sizeof(motion_data.ang_ref[jt]) );
-                if ( p_rb_tbl[cont_no]->arm_tbl[ano].jt_tbl[jt].type == TYPE_LINE )
-                {
-                    p_rb_tbl[cont_no]->arm_tbl[ano].jt_tbl[jt].home /= KHI_KRNX_M2MM;
-                }
-            }
-        }
-
-        fprintf( fp, ".JOINTS\n" );
-        for ( int ano = 0; ano < robot_info[cont_no].arm_num; ano++ )
-        {
-            fprintf( fp, "#rtchome%d", ano+1 );
-            for ( int jt = 0; jt < p_rb_tbl[cont_no]->arm_tbl[ano].jt_num; jt++ )
-            {
-                if ( p_rb_tbl[cont_no]->arm_tbl[ano].jt_tbl[jt].type == TYPE_LINE )
-                {
-                    fprintf( fp, " %.6f", p_rb_tbl[cont_no]->arm_tbl[ano].jt_tbl[jt].home*KHI_KRNX_M2MM );
-                }
-                else
-                {
-                    fprintf( fp, " %.6f", p_rb_tbl[cont_no]->arm_tbl[ano].jt_tbl[jt].home*180/M_PI );
-                }
-            }
-            fprintf( fp, "\n" );
-        }
-        fprintf( fp, ".END\n" );
-        fclose( fp );
-    }
-    else
+    if ( !makeRtcParam( cont_no, p_rb_tbl[cont_no]->robot_name.c_str(), param, sizeof(param), joint ) )
     {
-        setState( cont_no, ERROR );
```

Comment:
State transition to `ERROR` has disappeared.
Is it OK?

Reply:
It needs `setState( cont_no, ERROR )`.

Thanks.